ksql-parser/src/main/java/io/confluent/ksql/util/DataSourceExtractor.java

```diff
@@ -208,6 +136,81 @@ public Schema getFromSchema() {
     return fromSchema;
   }
 
+  private final class Visitor extends SqlBaseBaseVisitor<Node> {
+
+    @Override
+    public Node visitQuerySpecification(final SqlBaseParser.QuerySpecificationContext ctx) {
+      visit(ctx.from);
+      return visitChildren(ctx);
+    }
+
+    @Override
+    public Node visitTableName(final SqlBaseParser.TableNameContext context) {
+      return new Table(getLocation(context), getQualifiedName(context.qualifiedName()));
+    }
+
+    @Override
+    public Node visitAliasedRelation(final SqlBaseParser.AliasedRelationContext context) {
+      final Table table = (Table) visit(context.relationPrimary());
+
+      String alias = null;
```

Comment:
maybe extract this block into a method. Also, if `alias` is supposed to be non-null (see below) then maybe we should throw if it is `null`

Reply:
You're commenting on existing code. All I've done is move it.